### PR TITLE
update ffmpeg windows wrapper

### LINKED ISSUE #4890: Erroneous frame seek in VideoCapture::set (CAP_PROP_POS_FRAMES/CAP_PROP_POS_MSEC)
Transferred from http://code.opencv.org/issues/4081
```
|| Kesh Ikuma on 2014-12-22 22:14
|| Priority: Normal
|| Affected: branch 'master' (3.0-dev)
|| Category: highgui-video
|| Tracker: Bug
|| Difficulty: 
|| PR: 
|| Platform: x64 / Windows
```

Erroneous frame seek in VideoCapture::set (CAP_PROP_POS_FRAMES/CAP_PROP_POS_MSEC)
-----------
```
Setting either CAP_PROP_POS_FRAMES or CAP_PROP_POS_MSEC property does not always result in correct frame. I am using the prebuilt binary (x64/vc12/opencv_world300d.dll) in OpenCV 3.0-beta release. The issue can be observed in the following code snippet:

Mat rgb_frame;
VideoCapture cap("C:\myvideo.mp4"); 
cap.set(CAP_PROP_POS_FRAMES,503); // goto Frame 503
fprintf(0,"current frame: %f\n",cap.get(CAP_PROP_POS_FRAMES); // returns 503
cap.read(rgb_frame); // returns Frame 500

The behavior is consistent with both CAP_PROP_POS_FRAMES and CAP_PROP_POS_MSEC properties.

For the video file that I'm analyzing (FFmpeg stream info - Video: h264 (High) (avc1 / 0x31637661), yuv420p, 128x512, 225 kb/s, 30 fps, 30 tbr, 10000k tbn, 60 tbc) the frame is set correctly up to Frame 502, but when it is set to Frame 503, the subsequent read operation retrieves Frame 500 (3 frames earlier) although the get function returns 503. The offset is not consistent thereafter. Here is the summary of the observed offsets up to the first 3000 frames:

Frame Range  Offset
0-502        0
503-1443     3
1444-1646    4
1647-2888    3
2889-3000    0   (non-zero offset shows up again in the later frames)
```

History
-------
##### Maksim Shabunin on 2015-04-27 09:12
```
-   Target version changed from 3.0-beta to 3.0
```

### PR DESCRIPTION
**Pull request by alalek**
- ffmpeg 2.7.1=>3.1.1
- openh264 1.4.0=>1.5.0
- vpx 1.5.0 ([PR](https://github.com/opencv/opencv_3rdparty/pull/7))

resolves #4890

### CODE REVIEW

**Comment by alalek** [comment]
:+1:

File: modules/videoio/test/test_ffmpeg.cpp [hunk 1]
```diff
@@ -118,6 +118,9 @@ class CV_FFmpegWriteBigVideoTest : public cvtest::BaseTest
                     frame_s = Size(352, 288);
                 else if( tag == VideoWriter::fourcc('H', '2', '6', '3') )
                     frame_s = Size(704, 576);
+                else if( tag == VideoWriter::fourcc('H', '2', '6', '4') )
+                    // OpenH264 1.5.0 has resolution limitations, so lets use DCI 4K resolution
```
Comment (PhilLab) [inline modules/videoio/test/test_ffmpeg.cpp]: @alalek Out of curiosity: what kind of resolution limitations? A maximum in resolution or only specific width/height combinations?

Comment (alalek) [inline modules/videoio/test/test_ffmpeg.cpp]: @PhilLab I saw warning about these limitations:
width <= 4096
height <= ~2300

Comment (PhilLab) [inline modules/videoio/test/test_ffmpeg.cpp]: @alalek: Okay, thanks. Good to know

**Comment by peters** [comment]
@alalek I'd recommend upgrading libvpx to 1.6.0. Encoder speed has been greatly improved. I maintain my own private build of opencv-ffmpeg and on my Skylake CPU I'm observing a speed increase between 40-50%. Average video size on my video benchmark suite has increased by 35% on average though.

**Pull request merged**